pkg/currency/provider_fxratesapi.go

```diff
@@ -0,0 +1,109 @@
+package currency
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"github.com/justtrackio/gosoline/pkg/http"
+	"github.com/justtrackio/gosoline/pkg/log"
+)
+
+const FxRatesApiProviderName = "fxratesapi"
+
+type FxRatesApiResponse struct {
+	Success   bool               `json:"success"`
+	Timestamp int64              `json:"timestamp"`
+	Date      string             `json:"date"`
+	Base      string             `json:"base"`
+	Rates     map[string]float64 `json:"rates"`
+}
+
+func FxRatesApiProviderOption() ProviderOptions {
+	return func(ctx context.Context, logger log.Logger, http http.Client, settings ProviderSettings) (Provider, error) {
+		if settings.ApiKey == "" {
+			return nil, fmt.Errorf("no fxratesapi provider api key set")
+		}
+
+		logger.Info("using fxratesapi as currency provider")
+
+		return &fxRatesApiProvider{logger, http, &settings}, nil
+	}
+}
+
+type fxRatesApiProvider struct {
+	logger   log.Logger
+	http     http.Client
+	settings *ProviderSettings
```

Comment:
Why are the settings a pointer? I would make them not a pointer so we know they are never nil